internal/handlers/common/compare.go

```diff
@@ -248,8 +249,22 @@ func compareOrdered[T constraints.Ordered](a, b T) compareResult {
 }
 
 // compareNumbers compares two numbers.
-//
-// TODO https://github.com/FerretDB/FerretDB/issues/371
 func compareNumbers(a float64, b int64) compareResult {
-	return compareOrdered(a, float64(b))
+	if math.IsNaN(a) {
+		return notEqual
+	}
+
+	bigFloat := new(big.Float).SetFloat64(a).SetPrec(100000)
```

Comment:
Is that precision arbitrary? If yes, please add a comment something like:
```go
// TODO figure out correct precision
```

Reply:
Fixed